### ISSUE TITLE
[LAS 1.4] DLL doesn't read extended properties of point type 6 and above

### ISSUE THREAD

**Issue by quirysse** [LASzip/LASzip#15]
The problem is: extended point (types 6 and over) are not read correctly. The first part of the point is correct (`X`, `Y`, `Z`, `intensity`, etc. until `point_source_ID`) but all the extended attributes are wrong. 

The problem is caused by the layout difference between LAStempWritePoint10 and laszip_point. It seems that in the file _lasreaditemraw.hpp_, the `LASreadItemRaw_POINT14_LE` class is casting pointers to `LAStempWritePoint10` while `laszip_read_point()` is filling a `laszip_point` structure. 

The two structures share the same first fields but are different when it comes to the extended properties. Hence, it is not possible to read points that have extended classification attribute,  gps_time is wrong, and so on.

Comparing the two struct/class

``` c++
typedef struct laszip_point
{
/////////////// Common part
  laszip_I32 X;
  laszip_I32 Y;
  laszip_I32 Z;
  laszip_U16 intensity;
  laszip_U8 return_number : 3;
  laszip_U8 number_of_returns_of_given_pulse : 3;
  laszip_U8 scan_direction_flag : 1;
  laszip_U8 edge_of_flight_line : 1;
  laszip_U8 classification;
  laszip_I8 scan_angle_rank;
  laszip_U8 user_data;
  laszip_U16 point_source_ID;

/////////////// Here is the fork
  laszip_F64 gps_time;
  laszip_U16 rgb[4];
  laszip_U8 wave_packet[29];

  // LAS 1.4 only
  laszip_U8 extended_point_type : 2;
  laszip_U8 extended_scanner_channel : 2;
  laszip_U8 extended_classification_flags : 4;
  laszip_U8 extended_classification;
  laszip_U8 extended_return_number : 4;
  laszip_U8 extended_number_of_returns_of_given_pulse : 4;
  laszip_I16 extended_scan_angle;

  laszip_I32 num_extra_bytes;
  laszip_U8* extra_bytes;

} laszip_point_struct;
```

``` c++
class LAStempReadPoint10
{
/////////////// Common part
public:
  I32 x;
  I32 y;
  I32 z;
  U16 intensity;
  U8 return_number : 3;
  U8 number_of_returns_of_given_pulse : 3;
  U8 scan_direction_flag : 1;
  U8 edge_of_flight_line : 1;
  U8 classification;
  I8 scan_angle_rank;
  U8 user_data;
  U16 point_source_ID;

/////////////// Here is the fork

  // LAS 1.4 only
  I16 extended_scan_angle;
  U8 extended_point_type : 2;
  U8 extended_scanner_channel : 2;
  U8 extended_classification_flags : 4;
  U8 extended_classification;
  U8 extended_return_number : 4;
  U8 extended_number_of_returns_of_given_pulse : 4;
  // for 8 byte alignment of the GPS time
  U8 dummy[3];
  // LASlib only
  U32 deleted_flag;
  F64 gps_time;
};
```

**Comment by quirysse** [comment]
Actually, the same applies to the `LASwriteItemRaw_POINT14_LE`: when writing point type 6 or 7, GPS time is corrupted due to cast to incompatible structures.

**Comment by quirysse** [comment]
This has been fixed on https://github.com/quirysse/LASzip

**Comment by hobu** [comment]
LAZ 1.4 support is now available in the LASzip 3.1.0 release.
